src/pycram/datastructures/partial_designator.py

```diff
@@ -0,0 +1,113 @@
+    and generate a list of specified designators with all possible permutations of the input arguments.
+
+    PartialDesignators are designed as generators, as such they need to be iterated over to yield the possible specified
+    designators. Please also keep in mind that at the time of iteration all parameter of the specified designator_description need
+    to be filled, otherwise a TypeError will be raised, see the example below for usage.
+
+    .. code-block:: python
+            # Example usage
+            partial_designator = PartialDesignator(PickUpActionPerformable, milk_object_designator, arm=[Arms.RIGHT, Arms.LEFT])
+            for performable in partial_designator(Grasp.FRONT):
+                performable.perform()
+    """
+    performable: Type[ActionDesignatorDescription.Action]
+    """
+    Reference to the performable class that should be initialized
+    """
+    args: Tuple[Any, ...]
+    """
+    Arguments that are passed to the performable
+    """
+    kwargs: Dict[str, Any]
+    """
+    Keyword arguments that are passed to the performable
+    """
+
+    def __init__(self, performable: Type[ActionDesignatorDescription.Action], *args, **kwargs):
+        self.performable = performable
+        # Remove None values fom the given arguments and keyword arguments
+        self.args = tuple(filter(None, args))
+        self.kwargs = {k:v for k,v in kwargs.items() if v is not None}
+
+    def __call__(self, *fargs, **fkwargs):
+        """
+        Creats a new PartialDesignator with the given arguments and keyword arguments added.
+
+        :param fargs: Additional arguments that should be added to the new PartialDesignator
+        :param fkwargs: Additional keyword arguments that should be added to the new PartialDesignator
+        :return: A new PartialDesignator with the given arguments and keyword arguments added
+        """
+        newkeywords = {**self.kwargs, **fkwargs}
+        return PartialDesignator(self.performable, *self.args, *fargs, **newkeywords)
+
+    def __iter__(self) -> Type[ActionDesignatorDescription.Action]:
+        """
+        Iterates over all possible permutations of the arguments and keyword arguments and creates a new performable
+        object for each permutation.
+
+        :return: A new performable object for each permutation of arguments and keyword arguments
+        """
+        for args_combination in PartialDesignator.generate_permutations(self.args):
+            for kwargs_combination in PartialDesignator.generate_permutations(self.kwargs.values()):
+                yield self.performable(*args_combination, **dict(zip(self.kwargs.keys(), kwargs_combination)))
+
+    @staticmethod
+    def generate_permutations(args: Iterable) -> List:
+        """
+        Generates all possible permutations of the given arguments. This uses itertools.product to generate the
+        permutations.
```

Comment:
unnescessary information in the docstring

Reply:
write cartesian product or something that indicates what you are doing conceptually. If i want to know how you do it in detail i can read the code